scripts/update_source_data_schema_changelog.py

```diff
@@ -0,0 +1,198 @@
+    if not os.path.exists(directory):
+        sys.exit(f"Directory {directory} does not exist.")
+    return os.listdir(directory)
+
+
+def compare_lists(old_list=[], new_list=[]):
+    old_set = set(old_list)
+    new_set = set(new_list)
+
+    common = old_set & new_set
+    added = new_set - old_set
+    deleted = old_set - new_set
+
+    return list(common), list(added), list(deleted)
+
+
+def get_mapped_schema_json(directory: str, schema_name: str) -> {}:
+    """
+    Returns schema object indexed by field name.
+    Example:
+    {
+        "field_name_1": {
+            "name": field_name_1,
+            "type": STRING
+        },
+        "field_name_2": {
+            "name": field_name_2,
+            "type": STRING
+        },
+    }
+    """
+    schema_json = read_json_file(os.path.join(directory, schema_name))
+    schema_json_by_col_name = {column["name"]: column for column in schema_json}
+    return schema_json_by_col_name
+
+
+def compare_schemas(schemas=[]) -> {}:
+    schema_changes = {}
+
+    for schema in schemas:
+        old_schema_json_by_col_name = get_mapped_schema_json(
+            directory=OLD_SCHEMAS_DIR, schema_name=schema
+        )
+        new_schema_json_by_col_name = get_mapped_schema_json(
+            directory=NEW_SCHEMAS_DIR, schema_name=schema
+        )
+
+        common, added, deleted = compare_lists(
+            old_list=old_schema_json_by_col_name.keys(),
+            new_list=new_schema_json_by_col_name.keys(),
+        )
+
+        type_changed = [
+            (
+                col,
+                old_schema_json_by_col_name[col]["type"],
+                new_schema_json_by_col_name[col]["type"],
+            )
+            for col in common
+            if old_schema_json_by_col_name[col]["type"]
+            != new_schema_json_by_col_name[col]["type"]
+        ]
+
+        if added or deleted or type_changed:
+            schema_changes[schema] = {
+                "column_added": added,
+                "column_removed": deleted,
+                "type_changed": type_changed,
+            }
+
+    return schema_changes
+
+
+def get_print_label_string(label: str) -> str:
+    return f"\n## {label}:\n" if label else label
+
+
+def get_print_schemas_string(label="", schemas=[]) -> str:
+    print_string = ""
+    if not len(schemas):
+        return print_string
+
+    print_string += get_print_label_string(label)
+    for schema in schemas:
+        print_string += f"- {schema.replace('_schema.json', '')}"
+    return print_string
+
+
+def get_print_schema_changes_string(label="", schema_changes={}) -> str:
+    print_string = ""
+    if not schema_changes:
+        return print_string
+
+    print_string += get_print_label_string(label)
+
+    markdown_table = "|       Table Name                | Operation     | Columns                  |\n"
+    markdown_table += "|---------------------------------|---------------|--------------------------|\n"
```

Comment:
```suggestion
    markdown_table = "|       Table Name                | Operation     | Columns                                            |\n"
    markdown_table += "|---------------------------------|---------------|----------------------------------------------------|\n"
```

nit: make the `markdown_table` column sizes consistent